google/cloud/sql/connector/connector.py

```diff
@@ -109,13 +110,24 @@ def __init__(
             )
         self._instances: Dict[str, Instance] = {}
 
+        # initialize credentials
+        scopes = ["https://www.googleapis.com/auth/sqlservice.admin"]
+        if credentials:
+            if not isinstance(credentials, Credentials):
```

Comment:
Do we do this because there are other credential-like objects out there?

Reply:
This is the base class for credentials, there are several child types that can be supported so we just are checking that its atleast the base class

Reply:
Adding that as a comment might be nice.

Reply:
done